pkg/serverless/trace/inferredspan/propagation.go

```diff
@@ -0,0 +1,183 @@
+package inferredspan
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/aws/aws-lambda-go/events"
+)
+
+const (
+	awsTraceHeader = "AWSTraceHeader"
+	datadogHeader  = "_datadog"
+
+	rootPrefix    = "Root="
+	parentPrefix  = "Parent="
+	rootPadding   = len(rootPrefix + "1-00000000-00000000")
+	parentPadding = len(parentPrefix)
+)
+
+var rootRegex = regexp.MustCompile("Root=1-[0-9a-fA-F]{8}-00000000[0-9a-fA-F]{16}")
+
+type rawTraceContext struct {
+	TraceID  string `json:"x-datadog-trace-id"`
```

Comment:
We need sample priority too, right?

Reply:
Yes agreed. We didn't already support it for the existing SQS trace propagation via `x-datadog-*` headers, so I opted not to include it here. I'll file a separate ticket and we can add sample priority support for both header times in a single PR.

Reply:
Okay, thanks!